server-protocol.md

```diff
@@ -92,6 +85,26 @@ following keys (and ignores all others):
   future version of the protocol requires a rate-limiting CAPTCHA ticket or
   other authorization record, the server can send `error` (explaining the
   requirement) if it does not see this ticket arrive before the `bind`.
+* `permission-required`: a set of available authentication methods,
+  proof of work challenges etc.  The client needs to "solve" one of
+  them in order to get access to the service.
+
+The client should examine the `permissions-required` methods (if
+present) and select one to use (see also "Permission to Use the
+Server" below).  If the client doesn't support any permissions methods
+or has already chosen not to use any then it _may_ send `bind`
+immediately upon connecting. Older clients that do not know about
+`permsision-required` will always send the `bind` immediately (without
+waiting for the `welcome` message).
```

Comment:
```suggestion
waiting for the `welcome` message). In that case, the server MAY grant usage permission 
regardless. Otherwise, it will send a "permission denied" error message and close the connection.
```